hi-ml-histopathology/testhisto/testhisto/models/test_deepmil.py

```diff
@@ -255,38 +314,37 @@ def test_container(container_type: Type[LightningContainer], use_gpu: bool) -> N
 
     data_module: TilesDataModule = container.get_data_module()  # type: ignore
     data_module.max_bag_size = 10
+
     module = container.create_model()
+    module.trainer = MagicMock(world_size=1)  # type: ignore
+    module.log = MagicMock()  # type: ignore
     if use_gpu:
         module.cuda()
 
-    train_data_loader = data_module.train_dataloader()
-    for batch_idx, batch in enumerate(train_data_loader):
-        batch = move_batch_to_expected_device(batch, use_gpu)
-        loss = module.training_step(batch, batch_idx)
-        loss.retain_grad()
-        loss.backward()
-        assert loss.grad is not None
-        assert loss.shape == ()
-        assert isinstance(loss, Tensor)
-        break
+    assert_train_step(module, data_module, use_gpu)
+    assert_validation_step(module, data_module, use_gpu)
+    assert_test_step(module, data_module, use_gpu)
 
-    val_data_loader = data_module.val_dataloader()
-    for batch_idx, batch in enumerate(val_data_loader):
-        batch = move_batch_to_expected_device(batch, use_gpu)
-        outputs_dict = module.validation_step(batch, batch_idx)
-        loss = outputs_dict[ResultsKey.LOSS]  # noqa
-        assert loss.shape == ()  # noqa
-        assert isinstance(loss, Tensor)
-        break
 
-    test_data_loader = data_module.test_dataloader()
-    for batch_idx, batch in enumerate(test_data_loader):
-        batch = move_batch_to_expected_device(batch, use_gpu)
-        outputs_dict = module.test_step(batch, batch_idx)
-        loss = outputs_dict[ResultsKey.LOSS]  # noqa
-        assert loss.shape == ()
-        assert isinstance(loss, Tensor)
-        break
+@pytest.mark.parametrize("use_gpu", [True, False])
```

Comment:
I would parametrize over a few parameters of MockDeepSMILE, for example cache_mode, cache_location, is_finetune. Also encoder_type might be useful but I think it would add to much scope to this PR

Reply:
I tried running that locally, but it crashed my VM (too many cominations). I don't know on which nodes the CI runs but it might make it super slow. We might need to parametrize more but not all of them.